modules/ensemble/lib/framework/device.dart

```diff
@@ -13,22 +14,95 @@ import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
 import 'package:get_it/get_it.dart';
 import 'package:intl/intl.dart';
 
-/// get device information as well as requesting device permissions
-class Device
-    with
-        Invokable,
-        MediaQueryCapability,
-        LocationCapability,
-        DeviceInfoCapability {
+class Device with Invokable, MediaQueryCapability, LocationCapability, DeviceInfoCapability, WidgetsBindingObserver {
```

Comment:
Setup your formatter to property format code so it is consistent with everyone else. Your review reformats a lot of the code and it shows more changes than needed